tests/ui/bool_to_int_with_if.rs

```diff
@@ -123,3 +124,22 @@ fn side_effect() {}
 fn cond(a: bool, b: bool) -> bool {
     a || b
 }
+
+enum Enum {
+    A,
+    B,
+}
+
+fn if_let(a: Enum, b: Enum) {
+    if let Enum::A = a {
+        1
+    } else {
+        2
+    };
+
```

Comment:
These examples should trigger the lint, as it only detects `if <expr> {1} else {0}` AFAIK.

Could you update the example, to use these literals? :upside_down_face:

Reply:
Whoops 😄